File: app/src/main/java/com/tompee/utilities/filldevicespace/view/custom/NonSwipeablePager.java
package com.tompee.utilities.filldevicespace.view.custom;

import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

public class NonSwipeablePager extends ViewPager {
    private boolean mIsSwipeAllowed;

    public NonSwipeablePager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        return !mIsSwipeAllowed && super.onTouchEvent(event);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent event) {
        return !mIsSwipeAllowed && super.onInterceptTouchEvent(event);
    }

    public void interceptSwipeEvent(boolean intercept) {
        mIsSwipeAllowed = intercept;
    }
}

File: app/src/main/java/com/tompee/utilities/filldevicespace/feature/widget/NonSwipeablePager.kt
package com.tompee.utilities.filldevicespace.feature.widget

import android.content.Context
import android.support.v4.view.ViewPager
import android.util.AttributeSet
import android.view.MotionEvent

class NonSwipeablePager(context: Context, attrs: AttributeSet) : ViewPager(context, attrs) {

    var isSwipeAllowed: Boolean = false

    override fun onTouchEvent(event: MotionEvent?): Boolean {
        return !isSwipeAllowed && super.onTouchEvent(event)
    }

    override fun onInterceptTouchEvent(event: MotionEvent?): Boolean {
        return !isSwipeAllowed && super.onInterceptTouchEvent(event)
    }
}